pyznap/send.py

```diff
@@ -115,7 +115,7 @@ def send_filesystem(source_fs, dest_name, ssh_dest=None, raw=False, resume=False
 
     # get snapshots on source, catch exception if dataset was destroyed since pyznap was started
     try:
-        snapshots = source_fs.snapshots()[::-1]
+        snapshots = list(filter(lambda snap:snap_pred(snap.name.split('@')[1]), source_fs.snapshots()[::-1]))
```

Comment:
This doesn't filter the actual snapshots to send. This list is then used to find a base and most recent snapshot, which are then passed to `zfs send -I`, which handles all the intermediate snapshots.